background.ts

```diff
@@ -952,20 +952,39 @@ function validateEarlySettings(cfg: settings.Settings, newSettings: RecursivePar
  * The `requestShutdown` method is a special case that never returns.
  */
 class BackgroundCommandWorker implements CommandWorkerInterface {
-  protected k8sVersions: string[] = [];
   protected settingsValidator = new SettingsValidator();
 
   /**
    * Use the settings validator to validate settings after doing any
    * initialization.
    */
   protected async validateSettings(existingSettings: settings.Settings, newSettings: RecursivePartial<settings.Settings>) {
-    if (this.k8sVersions.length === 0) {
-      this.k8sVersions = (await k8smanager.kubeBackend.availableVersions).map(entry => entry.version.version);
-      this.settingsValidator.k8sVersions = this.k8sVersions;
+    let clearVersionsAfterTesting = false;
+
+    if (newSettings.kubernetes?.version && this.settingsValidator.k8sVersions.length === 0) {
+      // If we're starting up (by running `rdctl start...`) we probably haven't loaded all the k8s versions yet.
+      // We don't want to verify the proposed version makes sense (if it doesn't, we'll assign the default version later).
+      // Here we just want to make sure that if we're changing the version to a different value from the current one,
+      // the field isn't locked.
+      let currentK8sVersions = (await k8smanager.kubeBackend.availableVersions).map(entry => entry.version.version);
+
+      if (currentK8sVersions.length === 0 || (currentK8sVersions.length === 1 && currentK8sVersions[0] === '0.0.0')) {
```

Comment:
I don't understand the second part with `0.0.0.0`; I understand that you mentioned in one of the standups but there wasn't enough detail here (needs a comment pinpointing the cause, or better, eliminate it and whatever caused it).

Reply:
I can't repro it so I'll get rid of it.

Reply:
The "0.0.0" fake version comes from the mock server. If I set its `availableVersions` field to `[]` tests pass, and I don't need to special-case for `[SemVer('0.0.0')]`